internal/testutils/seed.go

```diff
@@ -0,0 +1,21 @@
+package testutils
+
+import (
+	"fmt"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+var randSeed struct {
+	value int64
+	once  sync.Once
+}
+
+func Seed() rand.Source {
```

Comment:
Why not return `*rand.Rand` here? That would make callers a bit less noisy.
```suggestion
func Rand() *rand.Rand {
```

Reply:
I agree. Keeping the complexity in the function is much cleaner.